File: customview/src/main/java/dinson/customview/weight/recycleview/CommonAdapter.java
package dinson.customview.weight.recycleview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * 单一布局通用数据适配器
 */
public abstract class CommonAdapter<T> extends RecyclerView.Adapter<CommonViewHolder>   {
    private final String TAG = "LGRecycleViewAdapter";
    //存储监听回调

    public List<T> mDataList;
    public Context mContext;


    public interface ItemClickListener {
        void onItemClicked(View view, int position);
    }

    public CommonAdapter(Context context, List<T> dataList) {
        this.mDataList = dataList;
        this.mContext = context;
    }

    /**
     * 获取列表控件的视图id(由子类负责完成)
     *
     * @param viewType
     * @return
     */
    public abstract int getLayoutId(int viewType);

    //更新itemView视图(由子类负责完成)
    public abstract void convert(CommonViewHolder holder, T t, int position);

    public T getItem(int position) {
        if (mDataList == null || position < 0 || mDataList.size() <= position)
            return null;
        return mDataList.get(position);
    }

    @Override
    public CommonViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        int layoutId = getLayoutId(viewType);
        CommonViewHolder viewHolder = CommonViewHolder.getViewHolder(parent, layoutId);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(CommonViewHolder holder, final int position) {
        T itemModel = getItem(position);
        convert(holder, itemModel, position);//更新itemView视图
    }

    @Override
    public int getItemCount() {
        if (mDataList == null)
            return 0;
        return mDataList.size();
    }

    public void destroyAdapter() {
        if (mDataList != null)
            mDataList.clear();
        mDataList = null;
    }

}

File: blingbase/src/main/java/com/dinson/blingbase/widget/recycleview/CommonAdapter.kt
package com.dinson.blingbase.widget.recycleview

import android.util.SparseArray
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView

/**
 * 单一布局通用数据适配器
 */
abstract class CommonAdapter<T>(var mDataList: MutableList<T>) : RecyclerView.Adapter<CommonViewHolder>() {
    private val mViewHolder = SparseArray<CommonViewHolder>()

    /**
     * 获取列表控件的视图id(由子类负责完成)
     */
    abstract fun getLayoutId(viewType: Int): Int

    //更新itemView视图(由子类负责完成)
    abstract fun convert(holder: CommonViewHolder, bean: T, position: Int)

    private fun getItem(position: Int): T {
        return mDataList[position]
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommonViewHolder {
        val layoutId = getLayoutId(viewType)
        return CommonViewHolder.getViewHolder(parent, layoutId)
    }

    override fun onBindViewHolder(holder: CommonViewHolder, position: Int) {
        val itemModel = getItem(position)
        mViewHolder.put(position, holder)
        convert(holder, itemModel, position) //更新itemView视图
    }

    override fun getItemCount(): Int {
        return mDataList.size
    }

    fun getCommonViewHolder(position: Int): CommonViewHolder {
        return mViewHolder[position]
    }
}